docs/developers/instantiate-testnet.md

```diff
@@ -0,0 +1,239 @@
+if you want to experiment with your own Celestia Testnetwork or if you want
+to test out new features to build as a core developer.
+
+## Hardware Requirements
+
+The following hardware minimum requirements are recommended for running the
+validator node:
+
+* Memory: 8 GB RAM
+* CPU: Quad-Core
+* Disk: 250 GB SSD Storage
+* Bandwidth: 1 Gbps for Download/100 Mbps for Upload
+
+## Setup Dependencies
+
+The following tutorial is done on an Ubuntu Linux 20.04 (LTS) x64
+instance machine.
+
+Once you have setup your instance, ssh into the instance to begin setting up
+the box with all the needed dependencies in order to run your bridge node.
+
+First, make sure to update and upgrade the OS:
+
+```sh
+sudo apt update && sudo apt upgrade -y
+```
+
+These are essential packages that are necessary to execute many tasks like
+downloading files, compiling and monitoring the node:
+
+```sh
+sudo apt install curl tar wget clang pkg-config libssl-dev jq \
+build-essential git make ncdu -y
+```
+
+### Install Golang
+
+Golang will be installed on this machine in order for us to be able to build
+the necessary binaries for running the bridge node. For Golang specifically,
+it is needed to be able to compile Celestia Application.
+
+```sh
+ver="1.17.2"
+cd $HOME
+wget "https://golang.org/dl/go$ver.linux-amd64.tar.gz"
+sudo rm -rf /usr/local/go
+sudo tar -C /usr/local -xzf "go$ver.linux-amd64.tar.gz"
+rm "go$ver.linux-amd64.tar.gz"
+```
+
+Now we need to add the `/usr/local/go/bin` directory to `$PATH`:
+
+```sh
+echo "export PATH=$PATH:/usr/local/go/bin:$HOME/go/bin" >> $HOME/.bash_profile
+source $HOME/.bash_profile
+```
+
+To check if Go was installed correctly run:
+
+```sh
+go version
+```
+
+Output should be the version installed:
+
+```sh
+go version go1.17.2 linux/amd64
+```
+
+## Installation
+
+Run the following:
+
+```sh
+cd $HOME
+git clone https://github.com/celestiaorg/celestia-app.git
+cd celestia-app/
+git checkout tags/v0.4.0 -b v0.4.0
+make install
+```
+
+To check if the binary was successfully compiled you can run the binary
+using the `--help` flag:
+
+```sh
+celestia-appd --help
+```
+
+You should see a similar output:
+
+```sh
+Stargate CosmosHub App
+
+Usage:
+  celestia-appd [command]
+
+Available Commands:
+*
+*
+*
+Use "celestia-appd [command] --help" for more information about a command.
+```
+
+## Spin Up A Celestia Testnet
+
+If you want to spin up a quick testnet with your friends, you can follow these steps.
+Unless otherwise noted, every step must be done by everyone who wants to
+participate in this testnet.
+
+### Optional: Reset Working Directory
+
+If you have already initialized a working directory for `celestia-appd` in the past,
+you must clean up before reinitializing a new directory. You can do so by running
+the following command:
+
+```sh
+celestia-appd unsafe-reset-all
+```
+
+### Initialize A Working Directory
+
+Run the following command:
+
+```sh
+celestia-appd init [moniker] --chain-id [chain-id]
```

Comment:
Should make these environment variables so that users can copy-paste commands.

Reply:
Resolved in new commit